lib/storage_tables/attachable/changes/create_one.rb

```diff
@@ -27,9 +27,20 @@ def blob
 
         def save
           unless StorageTables::Blob.service.exist?(attachment.full_checksum)
-            raise StorageTables::ActiveRecordError, "File is not yet uploaded"
+            raise StorageTables::ActiveRecordError,
+                  "No file exists with checksum #{attachment.full_checksum}, try uploading the file first. " \
+                  "Use the `attach` or `attachment=` method to upload the file."
           end
 
+          # Set the filename on the attachment
+          attachment.filename = filename
+
+          # Do not change anything if nothing has changed
+          return unless attachment.previous_changes.any? || attachment.changes.any?
+
+          # Delete the old attachment if it exists
+          attachment.class.where(record:).where.not(checksum: attachment.checksum).delete_all
+
           record.public_send(:"#{name}_storage_attachment=", attachment)
           record.public_send(:"#{name}_storage_blob=", blob)
```

Comment:
We kunnen nog de delete_all query naar de database vermijden als we de attachment gaan updaten. Als we in de attachments table voor has_one associations een unique index op de record_id hebben weten we zeker dat er geen ander attachment is voor hetzelfde record.

```suggestion
          if attachment.persisted?
            # Do not change anything if nothing has changed
            return if attachment.filename == filename

            # Set the filename on the attachment
            attachment.filename = filename
          else
            # Delete the old attachment if it exists
            attachment.class.where(record:).delete_all

            record.public_send(:"#{name}_storage_blob=", blob)
          end

          record.public_send(:"#{name}_storage_attachment=", attachment)
```